cpp/harvest/core.cc

```diff
@@ -54,10 +56,12 @@ namespace {
         "\t- input_dir: A dir with multiple JSON files to merge, typically from a search result.\n"
         "\t- output_file: The directory to store the merged JSON result file.\n"
         "\n"
-        "filter input_file output_file filtered_file\n"
+        "filter input_file output_file filtered_file keyword_file\n"
```

Comment:
1) if you introduce a filter_operator, you must also mention it in the line of the usage
2) Naming: keyword_file should be renamed to something like data_provider_ids_file, and since filter_operator only should affect the data_provider_ids_file, you should change the name to something closer to data_providers, e.g. data_provider_filter_type
3) The operator as well as the additional file should be treated as optional, so please change the implementation and put the names in brackets in the definition

example:
`filter input_file output_file filtered_file [data_provider_filter_type] [data_provider_ids_file]`